include/subgroup/tile/impl/fma_xe.hpp

```diff
@@ -24,6 +24,129 @@
+  static constexpr uint32_t b_block_size_y = matB_t::block_size_y;
+  static constexpr uint32_t b_block_elems = matB_t::block_elems;
+
+  static constexpr uint32_t tile_size_m = a_tile_size_y;
+  static constexpr uint32_t tile_size_k = a_tile_size_x;
+  static constexpr uint32_t tile_size_n = b_tile_size_x;
+  static constexpr uint32_t block_size_m = a_block_size_y;
+  static constexpr uint32_t block_size_k = a_block_size_x;
+  static constexpr uint32_t block_size_n = b_block_size_x;
+
+  static_assert(
+      a_tile_size_x == b_tile_size_y,
+      "matA tile k should match with matB tile k");
+  static_assert(
+      a_block_size_x == b_block_size_y,
+      "matA block k should match with matB block k");
+  static_assert(
+      b_block_size_y == matAcc_t::block_size_x,
+      "matA block k should match with matAcc block k");
+  static_assert(
+      b_block_size_x == matAcc_t::block_size_y,
+      "matb block n should match with matAcc block n");
+
+  static_assert(tile_size_m == 1, "matA tile m must be 1");
+  static_assert(a_block_size_y == 1, "matA block m must be 1");
+  __XETLA_API static void fma(
+      matAcc_t& dst,
+      matAcc_t& src,
+      matB_t& b,
+      matA_t& a) {
+#pragma unroll
+    for (uint32_t k = 0; k < tile_size_k / block_size_k; k++) {
+      auto a_block =
+          a.reg.xetla_select<block_size_m * block_size_k, 1>(k * block_size_k);
+#pragma unroll
+      for (uint32_t n = 0; n < tile_size_n / block_size_n; n++) {
+        uint32_t b_block_idx = n * tile_size_k / block_size_k + k;
+        auto b_block = b.reg.xetla_select<matB_t::block_elems, 1>(
+            b_block_idx * matB_t::block_elems);
+
+        uint32_t src_dst_idx = n * block_size_n;
+        auto src_block =
+            src.reg.xetla_select<block_size_n * block_size_k, 1>(src_dst_idx);
+        auto dst_block =
+            dst.reg.xetla_select<block_size_n * block_size_k, 1>(src_dst_idx);
+        fma_core<block_size_m, block_size_n, block_size_k>(
+            dst_block, src_block, b_block, a_block);
+      }
+    }
+  }
+  template <int blk_m, int blk_n, int blk_k>
+  __XETLA_API static void fma_core(
+      xetla_vector_ref<dtype_dst, blk_n * blk_k> __REF__ dst,
+      xetla_vector_ref<dtype_src, blk_n * blk_k> __REF__ src,
+      xetla_vector_ref<dtype_b, blk_k * blk_n> __REF__ b_block,
+      xetla_vector_ref<dtype_a, blk_m * blk_k> __REF__ a_block) {
+    static_assert(blk_m == 1, "block m must be 1");
+    auto dst_blk_2d = dst.xetla_format<dtype_dst, blk_n, blk_k>();
+    auto src_blk_2d = src.xetla_format<dtype_src, blk_n, blk_k>();
+    auto b_blk_2d = b_block.xetla_format<dtype_b, blk_n, blk_k>();
+    auto a_blk_2d = a_block.xetla_format<dtype_a, blk_m, blk_k>();
+#pragma unroll
+    for (uint32_t n = 0; n < blk_n; n++) {
+      dst_blk_2d.row(n) = b_blk_2d.row(n) * a_blk_2d.row(0) + src_blk_2d.row(n);
+    }
+  }
+  __XETLA_API static void reduce_acc_k(matAcc_t& matAcc, matDst_t_& matC) {
```

Comment:
reduction along K dimensions should be the default behavior

```suggestion
  __XETLA_API static void fma(matAcc_t& matAcc, matDst_t_& matC) {
```

Reply:
This step is executed at the end of the k loop.

Reply:
I try to keep it consistent with the original mma API